### PR TITLE
Recalibration tool

### PR DESCRIPTION
**Pull request by LachlanStuart**
This adds the recalibration command-line/library that I've been working on in #796 . That task has been shelved (reasons given in a comment on that issue). This project has already been [made public](https://pypi.org/project/msi-recal/) though not officially announced, and not integrated into METASPACE in any way.

The PyPI package has an awkward build script, because it vendorizes MSIWarp, which requires C++ code to be compiled. I did this so that all major OSs and Python versions could be supported - MSIWarp only has published packages for Win/macOS with Python 3.8. This PR includes a GitHub action for packaging & publishing to PyPI that handles all the OS/Python version combinations.

This PR also includes some molecular database files, as they are bundled into the PyPI package. The `matrix_` files were manually transcribed from [this paper](https://analyticalsciencejournals.onlinelibrary.wiley.com/doi/10.1002/rcm.9110). HMDB/CM3/LIPID_MAPS were just copied from the files that Ansible uses to install these DBs. Technically these could be downloaded during package compilation to avoid having them in Git, I was just lazy. It's up to you whether I should do this.

It's up to you how much reviewing this needs. As this code isn't used by METASPACE and isn't publicly advertised, I feel it's low-risk because if there were any bugs, they wouldn't have much impact.

### CODE REVIEW

File: .github/workflows/msi_recal_publish.yml [hunk 1]
```diff
@@ -1,13 +1,68 @@
...
+        path: metaspace/recal/MSIWarp
+        submodules: true
+
+    - uses: actions/setup-python@v2
+
+    - name: Build wheels
+      uses: joerick/cibuildwheel@v1.10.0
+      env:
+        CIBW_BUILD: "*_amd64 *_x86_64"
+        CIBW_SKIP: "cp27* cp35* pp*"
```
Comment (sergii-mamedov) [inline .github/workflows/msi_recal_publish.yml:30]: do we really want to support 2.7?

File: metaspace/recal/msi_recal/evaluate.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    #             for set_name, peaks_dict in self.collected_peak_sets.items():
+    #                 peaks = peaks_dict.get(mol_idx)
+    #                 if peaks is not None:
+    #                     peaks = peaks.loc[idxs]
+    #                     if len(peaks): WIP
+    #
+    #
+    #                 if mol_idx in peaks_dict:
+    #
+    #                 peaks = peaks_dict[mol_idx]
```
Comment (sergii-mamedov) [inline metaspace/recal/msi_recal/evaluate.py:116]: perhaps it should be removed if it is no longer used?

**Review by sergii-mamedov — APPROVED** [review]
LGTM

File: .github/workflows/msi_recal_publish.yml (hunk 1 above)
Comment (LachlanStuart) [reply .github/workflows/msi_recal_publish.yml:30]: `CIBW_SKIP` is the list of versions to exclude. Only [CPython 3.6-3.9](https://pypi.org/project/msi-recal/#files) are built with this combination of options.

**Pull request merged**